src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -277,7 +283,15 @@ ReactDOMComponent.Mixin = {
   mountComponent: function(rootID, transaction, context) {
     this._rootNodeID = rootID;
 
-    assertValidProps(this, this._currentElement.props);
+    var props = this._currentElement.props;
+
+    var wrapper = wrappers[this._tag];
+    if (wrapper) {
+      wrapper.mountWrapper.call(this, props, context);
```

Comment:
I find it hard to completely convince myself that complete inlining is guaranteed or even desirable in this case (due to branch prediction), but it seems plausible.

Maybe a case for macros? I just [tried Sweet.js](http://sweetjs.org/browser/editor.html#macro%20withWrapper%20%7B%0A%20%20%20%20rule%20%7B%20(%20$x:expr,%20($wrap:ident)%20=%3E%20%7B%20$body...%20%7D%20)%20%7D%20=%3E%20%7B%0A%20%20%20%20%20%20%20%20switch%20($x)%20%7B%0A%20%20%20%20%20%20%20%20%20%20case%20'input':%20var%20$wrap%20=%20ReactDOMInput;%20$body%20...%20break;%0A%20%20%20%20%20%20%20%20%20%20case%20'select':%20var%20$wrap%20=%20ReactDOMSelect;%20$body%20...%20break;%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%7D%0A%0AwithWrapper(this._tag,%20(wrapper)%20=%3E%20%7B%0A%20%20wrapper.mountWrapper(inst,%20props,%20context);%0A%20%20props%20=%20wrapper.getNativeProps(inst,%20props);%0A%7D);%0A) for the first time…

![image](https://cloud.githubusercontent.com/assets/6820/7904922/a8a6c6f2-07c2-11e5-8cab-8fee7ac9bd12.png)